commands/update.js

```diff
@@ -0,0 +1,110 @@
+import nativeCryptoLib from 'crypto'
+import chalk from 'chalk'
+import fetch from "cross-fetch"
+import Archethic, { Crypto, Utils } from 'archethic'
+import path from 'path'
+
+
+
+
+export async function handler({ files, endpoint, refIndex, refSeed }) {
+    console.log(chalk.blue('Fetching Reference Transaction'))
+    const onChainRefContent = requestRefTransaction(refIndex, refSeed, endpoint);
+    const localMetaData = await buildLocalRefContent(files);
+
+    const { unchangedMetaData, newFiles } = identifyModifiedFiles(localMetaData, (await onChainRefContent).metaData)
+    if (newFiles.length === 0) {
+        console.log(chalk.bgRedBright('No new files to upload'))
+        throw "No new files to upload";
+    }
+    files = files.filter(({ filePath }) => newFiles.includes(getFilePath(filePath)))
+    return { files: files, unchangedFilesMetadata: unchangedMetaData }
+}
+
+export function getNewRefContent(unchangedFilesMetadata, newRefContent) {
+    Object.assign(newRefContent.metaData, unchangedFilesMetadata);
+    return newRefContent
+}
+
+function identifyModifiedFiles(localMetaData, onChainMetaData) {
+    const unchangedMetaData = {};
+    const newFiles = [];
+    // put unchanged files in unchangedMetaData
+    Object.keys(localMetaData).forEach((pathAsKey) => {
+
+        if (onChainMetaData.hasOwnProperty(pathAsKey)) {
+            if (localMetaData[pathAsKey].hash === onChainMetaData[pathAsKey].hash) {
+                unchangedMetaData[pathAsKey] = onChainMetaData[pathAsKey];
+            }
+            else {
+                // client modification of file
+                newFiles.push(pathAsKey);
+            }
+        } else {
+            // client has added a new file 
+            newFiles.push(pathAsKey);
+        }
+    });
+    return { unchangedMetaData, newFiles };
+}
+
+async function buildLocalRefContent(files) {
+    return files.reduce((acc, { filePath, data }) => {
+        const hash = nativeCryptoLib.createHash('sha-1').update(data).digest('hex')
```

Comment:
'sha-1' is not supported on node v16, you should us 'sha1'

Reply:
it's throwing an error, you have to be consistent with the lib.js hash function

Reply:
Hello the earlier pr has been already merge https://github.com/archethic-foundation/aeweb-cli/issues/94
I have uncommneted try and catch,
please try to use node js version as was in .tools-version in aenode as v18